packages/web-components/fast-foundation/src/tree-view/tree-view.ts

```diff
@@ -20,223 +20,296 @@ import { FoundationElement } from "../foundation-element";
  * @public
  */
 export class TreeView extends FoundationElement {
-    public treeView: HTMLElement;
-
+    /**
+   /**
+    * When true, the control will be appear expanded by user interaction.
+    * @public
+    * @remarks
+    * HTML Attribute: render-collapsed-nodes
+    */
     @attr({ attribute: "render-collapsed-nodes" })
     public renderCollapsedNodes: boolean;
 
+    /**
+     * The currently selected tree item
+     * @public
+     */
     @observable
-    public currentSelected: HTMLElement | TreeItem | null;
-
-    @observable
-    private nested: boolean;
+    public currentSelected: HTMLElement | TreeItem | null = null;
+    private currentSelectedChanged(oldValue, newValue): void {
+        if (this.$fastController.isConnected) {
+            if (isTreeItemElement(oldValue)) {
+                (oldValue as TreeItem).selected = false;
+            }
+        }
+    }
 
+    /**
+     *  Slotted children
+     *
+     * @internal
+     */
     @observable slottedTreeItems: HTMLElement[];
-    private slottedTreeItemsChanged(oldValue, newValue): void {
+    private slottedTreeItemsChanged(): void {
         if (this.$fastController.isConnected) {
-            // filter the tree items until that's done for us in the framework
-            this.resetItems();
-            this.treeItems = this.getVisibleNodes();
+            // update for slotted children change
             this.setItems();
-
-            // check if any tree items have nested items
-            // if they do, apply the nested attribute
-            if (this.checkForNestedItems()) {
-                this.slottedTreeItems.forEach(node => {
-                    if (isTreeItemElement(node)) {
-                        (node as TreeItem).nested = true;
-                    }
-                });
-            }
         }
     }
 
-    private checkForNestedItems(): boolean {
-        return this.slottedTreeItems.some((node: HTMLElement) => {
-            return isTreeItemElement(node) && node.querySelector("[role='treeitem']");
-        });
-    }
+    /**
+     * The tree item that is designated to be in the tab queue.
+     *
+     * @internal
+     */
+    @observable
```

Comment:
Does this need to be observable? I don't see any observers or change handlers

Reply:
no, I'll take that out.